### PR TITLE
libc:machine:risc-v:unifying elf relocation code.

### PR DESCRIPTION
**Pull request by hotislandn**
Signed-off-by: hotislandn <hotislandn@hotmail.com>

## Summary
This patch merges the two arch_elf.c files dedicated for rv32 and rv64 into a unified one.

## Impact
libs:libc:machine for risc-v targets.

## Testing
rv32: verified on bl602evb:elf.
rv64: verified on QEMU with elf files.

### CODE REVIEW

File: libs/libc/machine/risc-v/common/arch_elf.c [hunk 1]
```diff
@@ -224,7 +246,7 @@ bool up_checkarch(FAR const Elf_Ehdr *ehdr)
 
   if ((ehdr->e_entry & 1) != 0)
     {
-      berr("ERROR: Entry point is not properly aligned: %08" PRIx32 "\n",
+      berr("ERROR: Entry point is not properly aligned: %08lx\n",
```
Comment (gustavonihei) [inline libs/libc/machine/risc-v/common/arch_elf.c:249]: As soon as #2863 is merged, this change won't be needed.
We should stick to using the definitions from `inttypes.h`

**Review by gustavonihei — CHANGES_REQUESTED** [review]

File: libs/libc/machine/risc-v/common/arch_elf.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/machine/risc-v/common/arch_elf.c:249]: #2863 has been merged, @hotislandn

Comment (hotislandn) [reply libs/libc/machine/risc-v/common/arch_elf.c:249]: PRIx32 works for RV32, but not for RV64:

machine/risc-v/common/arch_elf.c:241:12: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'Elf64_Addr' {aka 'const long unsigned int'} [-Wformat=]
       berr("ERROR: Entry point is not properly aligned: %08" PRIx32 "\n",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ehdr->e_entry);
            ~~~~~~~~~~~~~

This parameter, "ehdr->e_entry" is also 'const long unsigned int' for RV32, which can be tested via following:

machine/risc-v/common/arch_elf.c:241:12: warning: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'Elf32_Addr' {aka 'const long unsigned int'} [-Wformat=]
       berr("ERROR: Entry point is not properly aligned: %08" PRIxPTR "\n",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            ehdr->e_entry);
            ~~~~~~~~~~~~~

For "PRIx32", it is expanded to "lx" on RV32, but "x" on RV64; a similar issue happens for "PRIxPTR".

The reason for this change is that the input parameter has the same base type: 'const long unsigned int', which requires to use "lx" for printf, though it presents different data lengths on RV32 and RV64.

File: libs/libc/machine/risc-v/common/arch_elf.c [hunk 2]
```diff
@@ -42,6 +42,26 @@
...
 
+/* ELF32 and ELF64 definitions */
+
+#ifdef CONFIG_LIBC_ARCH_ELF_64BIT
+#  define ARCH_ELF_TYP_STR    "64"
+#  define ARCH_ELF_CLASS      ELFCLASS64
+#  define ARCH_ELF_RELTYPE    ELF64_R_TYPE
+#  define ARCH_ELF_HDR_STR    Elf64_Ehdr
+#  define ARCH_ELF_REL_STR    Elf64_Rel
+#  define ARCH_ELF_SYM_STR    Elf64_Sym
```
[the remark this answers is not in the document]
Comment (hotislandn) [reply libs/libc/machine/risc-v/common/arch_elf.c]: Patch updated: redundant code slices removed.

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM

**Pull request merged**